frontend/src/design/components/popovers/NotificationsPopover.js

```diff
@@ -101,6 +101,29 @@ export const NotificationsPopover = () => {
     }
   }, [client]);
 
+  const resolve_link = (notification) => {
+    if (notification.type === 'METADATA_FORM_UPDATE') {
+      const entity_type = notification.target_uri.split('|')[1].toLowerCase();
+      const entity_uri = notification.target_uri.split('|')[0];
+      const main_modules = [
+        'environment',
+        'organization',
+        'dataset',
+        'share',
+        'dashboard',
+        'worksheet',
+        'pipeline',
+        'notebook'
+      ];
+      if (main_modules.includes(entity_type)) {
+        return `/console/${entity_type}s/${entity_uri}`;
```

Comment:
for dataset the path is `/console/s3-datasets/:uri` or `/console/redshift-datasets/:uri` depending on the dataset type (for orgs or envs this should work i think)

Reply:
I separated s3 and rd datasets on backend. Added DB migration

Reply:
Are all these new changes needed / should we be splitting out S3 Datasets and Redsfhit Datasets to be Handled separately (i.e. as different entities)?

If all datasets will operate with Metadata Forms the same I would suggest we can just leverage `datasetType` that already exists in data.all base. Meaning `def create_metadata_form_version(...)`: would include

```
            all_attached = MetadataFormRepository.get_all_attached_metadata_forms(session, uri)
            for attached in all_attached:
                entity = MetadataFormService.resolve_attached_entity(attached)
                owner = MetadataFormService.get_entity_owner(entity)
 
                entityType = attached.entityType 
                if MetadataFormEntityTypes.Datasets.value == entity:
                     entityType = MetadataFormService._get_entity_type(entity)
                if owner:
                    NotificationRepository.create_notification(
                        ...
                        target_uri=f'{attached.entityUri}|{attached.entityType}',
                        ...
                    )
```

where `_get_entity_type` is:

```
    def _get_entity_type(entity):
        return f"{entity.datasetType}-Dataset"
```

The above may also help with scale if dataset types blows up to a large number don't need to map MF entityType to DatasetType 1-1

Reply:
I think it's better to separate them in database for the sake of enforcement. 
The customer may have different view on managing MF for redshift and S3.